### ISSUE TITLE
meson fails to detect strverscmp

### ISSUE THREAD

**Issue by bsolar17** [BestImageViewer/geeqie#1061]
### ISSUE TYPE
* Bug Report

### GEEQIE VERSION
```
Geeqie 2.0+git20230126-c01dcf42 GTK3
```

### OS / DISTRIBUTION
Linux debian 5.10.0-21-arm64

### SUMMARY

Build issue: configuration fails to detect `strverscmp`, causing the built application to lack the "sort by number" functionality.

This seems to be caused by `strverscmp` being a GNU extension, but meson not setting the `_GNU_SOURCE` macro when performing the detection:

https://github.com/BestImageViewer/geeqie/blob/c01dcf42abbac84f6ecb1919560c7d8339af47ac/meson.build#L503

I think should be:

```
result = cc.has_function('strverscmp', prefix : '#include <string.h>', args : '-D_GNU_SOURCE')
```

### STEPS TO REPRODUCE
1. Configure on a system with strverscmp available.
2. Check HAVE_STRVERSCMP in build/config.h

HAVE_STRVERSCMP should be defined, but it's not. With the fix mentioned above it gets defined correctly.

**Comment by caclark** [comment]
This sort option might be redundant.

On Edit/Preferences/File Filters is the option Natural Sort Order. This uses g_utf8_collate_key_for_filename() to achieve the same effect (more or less) as strverscmp() . (Changing  the option requires a restart of Geeqie to regenerate existing FileDatas).

I feel inclined to delete the Sort By Number option and wait for the complaints.

**Comment by bsolar17** [comment]
I tried Natural Sort Order before but didn't realize it required a restart of the application. After a restart it seems to work well enough, I noticed only minor differences in sorting with filenames with symbols in them.

**Comment by caclark** [comment]
@DrMcCoy 

Implemented in commit fcf71d9

**Comment by DrMcCoy** [comment]
Thank you! :)
